Internal Release Play Store App Milestone

komidawi opened the pull request.

Shino1024 commented on app/src/main/java/com/github/komidawi/pizzacostcalculator/MainActivity.java:
```diff
@@ -3,12 +3,64 @@
 public class MainActivity extends AppCompatActivity {
+    private EditText diagonalInput;
+    private EditText priceInput;
+    private TextView ratioDisplay;
```
Read about ButterKnife

komidawi replied:
It does not support `afterTextChanged`, so currently imo there is no benefit to using ButterKnife, moreover author himself says:
> Attention: Development on this tool is winding down. Please consider switching to view binding in the coming months.

Shino1024 replied:
You are right, I haven't followed ButterKnife's development for a while.

Shino1024 commented on app/src/main/java/com/github/komidawi/pizzacostcalculator/calculations/Calculator.java:
```diff
@@ -3,12 +3,15 @@
-    public static double calculateRatio(double diagonal, double price, PizzaShape pizzaShape) {
+    private static final int SQ_CENTIMETERS_IN_SQ_METRE = 10000;
+
+    // Calculates price / area ratio in currency units per one square meter of pizza
```
Do it in Java docs style.

komidawi replied:
Can I omit @params (as they are self-explanatory) and write only method description?

Shino1024 replied:
It is recommended to document everything, actually. It is a good practice and makes your code ever easier to read. That's actually something I am going to do in my future projects. :)

Shino1024 commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -2,17 +2,118 @@
+        android:id="@+id/price_input"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginStart="64dp"
```
Use dimen.xml.

komidawi committed:
Build: Configure project build options

komidawi committed:
[PCC-1] Build: Add viewBinding support

komidawi committed:
[PCC-1] Feat: Create first working Fragment

komidawi committed:
[PCC-2] Build: Add Navigation support

komidawi committed:
[PCC-2] Feat: Create Fragment Navigation

komidawi committed:
[PCC-3] Feat: Add navigation Up Button in ActionBar

komidawi committed:
[PCC-4] Feat: Create Title screen

komidawi committed:
[PCC-4] Refactor: Rename SecondFragment to HomeFragment

komidawi committed:
[PCC-6] Feat: Add 3dot Options Menu and About Fragment

komidawi committed:
[PCC-7] Build: Add Safe Args dependency

komidawi committed:
[PCC-7] Feat: Use Safe Args to pass data between Fragments

komidawi committed:
[PCC-8] Feat: Add Share Intent

komidawi committed:
[PCC-5] Build: Set compile options to JVM 1.8

komidawi committed:
[PCC-5] Feat: Add hamburger menu (DrawerLayout)

komidawi committed:
[PCC-9] Feat: Add animations between Fragments

komidawi committed:
[PCC-11] Dev: Add Lifecycle handling

komidawi committed:
[PCC-12] Dev: Save state when app is killed

komidawi committed:
[PCC-14] App: Reset project

komidawi committed:
[PCC-14] Build: Upgrade libraries, gradle, gradle plugin to latest ones

komidawi committed:
[PCC-14] Feat: Add new blank Activity and Fragment

komidawi committed:
[PCC-14] Feat: Display sample pizza data

komidawi committed:
[PCC-14] Refactor: Use common version variable in build.gradle

komidawi committed:
[PCC-14] Feat: Add create Pizza screen

komidawi committed:
[PCC-17] Feat: Add simple FAB on main screen

komidawi committed:
[PCC-19] Feat: Make back button return to previous Fragment

komidawi committed:
[PCC-20] Feat: Pass created Pizza between Fragments

komidawi committed:
[PCC-20] Fix: Pizzas not displaying on PizzaListFragment

komidawi committed:
[PCC-21] Build: Add Room and Lifecycle Extensions dependencies

komidawi committed:
[PCC-21] Feat: Create Pizza Entity

komidawi committed:
[PCC-21] Refactor: Remove unused old classes

komidawi committed:
[PCC-21] Feat: Create database Dao

komidawi committed:
[PCC-21] Feat: Create Room database

komidawi committed:
[PCC-21] Feat: Make ViewModels use database Dao

komidawi committed:
[PCC-21] Feat: Make PizzaListFragment use database

komidawi committed:
[PCC-21] Refactor: Restructure UI and data packages

komidawi committed:
[PCC-21] Feat: Make AddPizzaFragment use database

komidawi committed:
[PCC-21] Feat: Display Toast when required fields empty on adding Pizza

komidawi committed:
[PCC-25] Feat: Use RecyclerView to display Pizza list

komidawi committed:
[PCC-25] Feat: Add layout for Pizza list item

komidawi committed:
[PCC-25] Refactor: Encapsulate ViewHolder-creating related code

komidawi committed:
[PCC-25] Refactor: Use ListAdapter instead of RecyclerView.Adapter

komidawi committed:
[PCC-25] Refactor: Use DataBinding in RecyclerView

komidawi committed:
[PCC-25] Refactor: Add Binding Adapters

komidawi committed:
[PCC-25] Feat: Add click listener to RecyclerView

komidawi committed:
[PCC-25] Feat: Add Header to RecyclerView

komidawi committed:
[PCC-25] Refactor: Apply Coroutines to optimize submitting pizza list

komidawi committed:
[PCC-26] Feat: Improve look of Pizza List header

komidawi committed:
[PCC-27] Build: Upgrade library versions

komidawi committed:
[PCC-27] Refactor: Extract methods from Fragments, move ViewModelFactory

komidawi committed:
[PCC-27] Refactor: Remove unused code

komidawi committed:
[PCC-27] Feat: Add Fragment titles

komidawi committed:
[PCC-31] Feat: Improve Add Pizza layout

komidawi committed:
[PCC-33] Refactor: Remove unused code

komidawi committed:
[PCC-33] Refactor: Use BigDecimal for price and size fields

komidawi committed:
[PCC-28] Build: Add plethora of Testing dependencies

I spent 3 hours trying to make this working. In the end I just copied all the dependencies from Android Codelabs testing lesson.

komidawi committed:
[PCC-28] Test: Create first Add Pizza ViewModel unit test

komidawi committed:
[PCC-28] Test: Add input tests for adding Pizza

komidawi committed:
[PCC-28] Test: Refactor: Use extension function to observe LiveData

komidawi committed:
[PCC-28] Dev: Exclude .ignore directory from git

komidawi committed:
[PCC-28] Test: Add implementation to test DatabaseDao

komidawi committed:
[PCC-28] Test: Add TestCoroutineDispatcher

komidawi committed:
[PCC-28] Test: Add handleAddPizza() tests

komidawi committed:
[PCC-28] Test: Add more testing dependencies

komidawi committed:
[PCC-28] Test: Use JUnit Rule to set TestCoroutineDispatcher

komidawi committed:
[PCC-28] Test: Add PizzaDatabaseDao tests

komidawi committed:
[PCC-28] Test: Create Add Pizza navigation event test

komidawi committed:
[PCC-28] Build: Add test:core and test.ext:junit deps

komidawi committed:
[PCC-28] Test: Use ServiceLocator for Fragment DI, add instrumented test

komidawi committed:
[PCC-42] Build: Upgrade dexMakerVersion from 2.12.1 to 2.19.0

komidawi committed:
[PCC-42] Test: Add Espresso integration tests

komidawi committed:
[PCC-42] Test: Remove default sample tests

komidawi committed:
[PCC-43] Test: Create first E2E Test for Main Activity

komidawi committed:
[PCC-34] Feat: Add Remove Pizza button on Pizza List

komidawi committed:
[PCC-28] Test: Provide null and empty input Pizza data Tests

komidawi committed:
[PCC-28] Refactor: Test: Make  AddPizzaFragmentVM tests shorter

komidawi committed:
[PCC-28] Db: Enable destructive migration

komidawi committed:
Merge branch 'develop' into PCC-34-remove-button

komidawi committed:
Merge branch 'PCC-43-e2e-testing' into PCC-34-remove-button

komidawi committed:
[PCC-34] Test: Add remove Pizza button tests

komidawi committed:
[PCC-34] Feat: Add remove Pizza button header text

komidawi committed:
[PCC-34] Refactor: Extract Pizza list item styles

komidawi committed:
[PCC-45] Feat: Create basic Cost Calculator

komidawi committed:
[PCC-45] Feat: Integrate Calculator into ViewModel

komidawi committed:
[PCC-45] Feat: Add Ratio display when for AddPizzaFragment.kt

komidawi committed:
[PCC-45] Refactor: Use Factory for creating Pizza entities

komidawi committed:
[PCC-47] Build: Add CI-CD status to Readme

komidawi committed:
[PCC-47] Fix: Temporary: Increase sleep() time for failing tests

komidawi committed:
[PCC-47] Build: Add Code Coverage reports plugin

komidawi committed:
[PCC-47] Build: Add code coverage status to readme

komidawi committed:
Merge branch 'develop' into PCC-45-calculation

komidawi committed:
[PCC-45] Feat: Add ratio to UI

komidawi committed:
[PCC-45] Refactor: Change all Floats to Doubles

komidawi committed:
[PCC-45] Feat: Improve ratio display on create Pizza screen

komidawi committed:
[PCC-45] Fix: Failing tests for Calculator and AddPizzaFragment

komidawi committed:
Merge branch 'master' of github.com:komidawi/PizzaCostCalculator into develop

# Conflicts:
#	README.md

komidawi committed:
[PCC-65] Refactor: Global code cleanup

komidawi committed:
Merge pull request #2 from komidawi/PCC-65-code-cleanup

[PCC-65] Refactor: Global code cleanup
